core/src/main/scala/cats/Traverse.scala

```diff
@@ -30,6 +30,12 @@ import simulacrum.typeclass
     U.TC.traverse(fa)(a => U.subst(f(a)))(this)
 
   /**
+   * A traverse followed by flattening the inner result.
+   */
+  def traverseM[G[_], A, B](fa: F[A])(f: A => G[F[B]])(implicit G: Applicative[G], F: FlatMap[F]): G[F[B]] =
+    G.map(traverse(fa)(f))(F.flatten)
```

Comment:
@adelbertc The `Applicative[G]` is required on `traverse` also. Here I think of the `M` as "monadic", referring to the `FlatMap[F]`. I guess we could call it `traverseF` instead, though that runs the risk of being confused with `Functor`.